src/main/java/com/backbase/ct/bbfuel/client/user/UserPresentationRestClient.java

```diff
@@ -1,25 +1,37 @@
 package com.backbase.ct.bbfuel.client.user;
 
+import static com.backbase.ct.bbfuel.util.ResponseUtils.isBadRequestException;
+import static org.apache.http.HttpStatus.SC_CREATED;
 import static org.apache.http.HttpStatus.SC_OK;
 
 import com.backbase.ct.bbfuel.client.common.RestClient;
+import com.backbase.ct.bbfuel.client.legalentity.LegalEntityPresentationRestClient;
 import com.backbase.ct.bbfuel.config.BbFuelConfiguration;
+import com.backbase.integration.user.rest.spec.v2.users.UsersPostRequestBody;
+import com.backbase.presentation.user.rest.spec.v2.users.IdentitiesPostRequestBody;
 import com.backbase.presentation.user.rest.spec.v2.users.LegalEntityByUserGetResponseBody;
 import com.backbase.presentation.user.rest.spec.v2.users.UserGetResponseBody;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
 import javax.annotation.PostConstruct;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Component;
 
+@Slf4j
 @Component
 @RequiredArgsConstructor
 public class UserPresentationRestClient extends RestClient {
 
     private final BbFuelConfiguration config;
+    private final LegalEntityPresentationRestClient legalEntityPresentationRestClient;
```

Comment:
same here
`doesn't feel right, making one Rest Client dependent on the other one`